fe/fe-core/src/main/java/org/apache/doris/backup/Repository.java

```diff
@@ -241,10 +242,13 @@ public Status alterRepositoryS3Properties(Map<String, String> properties) {
 
     @Override
     public void gsonPostProcess() {
-        StorageBackend.StorageType type = StorageBackend.StorageType.BROKER;
-        if (!type.equals(StorageBackend.StorageType.BROKER)) {
+        try {
             StorageProperties storageProperties = StorageProperties.createPrimary(this.fileSystem.properties);
             this.fileSystem = FileSystemFactory.get(storageProperties);
+        } catch (StoragePropertiesException exception) {
+            LOG.warn("Failed to create file system for repository: {}, error: {}, roll back to broker"
+                    + " filesystem", name, exception.getMessage());
+            this.fileSystem = FileSystemFactory.get(this.fileSystem.name, this.fileSystem.properties);
         }
```

Comment:
We will refactor this interface within the unified FS Factory method. In fact, we need a way to determine which types qualify as BrokerFS.